mlrun/api/initial_data.py

```diff
@@ -494,33 +497,75 @@ def _enrich_project_state(
             db.store_project(db_session, project.metadata.name, project)
 
 
+def _delete_default_marketplace_source(db_session: sqlalchemy.orm.Session):
+    """
+    Delete default marketplace source directly from db
+    """
+    # Not using db.delete_marketplace_source() since it doesn't allow deleting the default marketplace source.
+    default_record = (
+        db_session.query(MarketplaceSource)
+        .filter(MarketplaceSource.index == mlrun.api.schemas.last_source_index)
+        .one_or_none()
+    )
+    if default_record:
+        logger.info(f"Deleting default marketplace source {default_record.name}")
+        db_session.delete(default_record)
+        db_session.commit()
+    else:
+        logger.info("Default marketplace source not found")
+
+
+def _update_default_marketplace_source(
+    db: mlrun.api.db.sqldb.db.SQLDB,
+    db_session: sqlalchemy.orm.Session,
+    overwrite: bool = False,
```

Comment:
It seems that the `overwrite` parameter is pretty redundant - you call this function in 2 places:
1. as part of the data-migration flow, where `overwrite=True`
2. as part of initial-data flow, where it's `False`, but you only call this if there isn't a record already.
Note also that calling this with `overwrite=False` where a record already exists will result in an error/exception from the DB due to duplicate primary key value.

So, I think you should remove this parameter, and *always* delete the record if it exists (i.e. always call `_delete_default_marketplace_source`)